File: make_layout.php
<?php
$headerPath = '/home/zorin/Descargas/public_html/app/views/layouts/header.php';
$footerPath = '/home/zorin/Descargas/public_html/app/views/layouts/footer.php';
$layoutPath = '/home/zorin/Descargas/penalito-laravel/resources/views/layouts/app.blade.php';

if (!is_dir(dirname($layoutPath))) {
    mkdir(dirname($layoutPath), 0777, true);
}

$header = file_get_contents($headerPath);
$footer = file_get_contents($footerPath);

$layout = $header . "\n\n        @yield('content')\n\n" . $footer;

// Convert <?php echo URLROOT; ?>/assets... to {{ asset('assets...') }}
$layout = str_replace('<?php echo URLROOT; ?>/assets/', "{{ asset('assets/') }}", $layout);
$layout = str_replace('<?php echo URLROOT; ?>/video/', "{{ asset('video/') }}", $layout);

// Convert <?php echo URLROOT; ?>/ to {{ url('/') }}/
$layout = str_replace('<?php echo URLROOT; ?>/', "{{ url('/') }}/", $layout);
$layout = str_replace('href="<?php echo URLROOT; ?>"', 'href="{{ url(\'/\') }}"', $layout);

// Replace title
$layout = str_replace('<?php echo isset($data[\'title\']) ? $data[\'title\'] : SITENAME; ?>', '{{ $title ?? config(\'app.name\') }}', $layout);

// Replace date('Y')
$layout = str_replace('<?php echo date(\'Y\'); ?>', '{{ date(\'Y\') }}', $layout);

file_put_contents($layoutPath, $layout);
echo "Layout created at $layoutPath\n";

File: make_layout.py
import os

header_path = '/home/zorin/Descargas/public_html/app/views/layouts/header.php'
footer_path = '/home/zorin/Descargas/public_html/app/views/layouts/footer.php'
layout_path = '/home/zorin/Descargas/penalito-laravel/resources/views/layouts/app.blade.php'

os.makedirs(os.path.dirname(layout_path), exist_ok=True)

with open(header_path, 'r', encoding='utf-8') as f:
    header = f.read()

with open(footer_path, 'r', encoding='utf-8') as f:
    footer = f.read()

layout = header + "\n\n        @yield('content')\n\n" + footer

# Replacements
layout = layout.replace('<?php echo URLROOT; ?>/assets/', "{{ asset('assets/') }}")
layout = layout.replace('<?php echo URLROOT; ?>/video/', "{{ asset('video/') }}")
layout = layout.replace('<?php echo URLROOT; ?>/', "{{ url('/') }}/")
layout = layout.replace('href="<?php echo URLROOT; ?>"', 'href="{{ url(\'/\') }}"')
layout = layout.replace("<?php echo isset($data['title']) ? $data['title'] : SITENAME; ?>", "{{ $title ?? config('app.name') }}")
layout = layout.replace("<?php echo date('Y'); ?>", "{{ date('Y') }}")

with open(layout_path, 'w', encoding='utf-8') as f:
    f.write(layout)

print("Layout created at " + layout_path)
